tests/PHPStan/Rules/Functions/IncompatibleDefaultParameterTypeRuleTest.php

```diff
@@ -24,6 +24,10 @@ public function testFunctions(): void
 				'Default value of the parameter #1 $string (false) of function IncompatibleDefaultParameter\takesString() is incompatible with type string.',
 				15,
 			],
+			[
+				'Default value of the parameter #1 $x (1) of function IncompatibleDefaultParameter\takesTemplateWithIntDefault() is incompatible with type T.',
```

Comment:
This issue actually shouldn't be reported. I suspect it's because TemplateTypeMixed isn't a supertype of FloatType.

Reply:
I don't know the solution currently, will have to think about it.

Reply:
I can drop the test if you want. I added it to make sure I didn't introduce any unintended side effects.

Reply:
No, this is something that actually needs fixing.